core/corehttp/hostname.go

```diff
@@ -266,6 +283,19 @@ func isPeerIDNamespace(ns string) bool {
 	}
 }
 
+// Converts an identifier to DNS-safe representation that fits in 63 characters
+func toDNSPrefix(ctx context.Context, api iface.CoreAPI, id string) (prefix string) {
+	// Return as-is if things fit
+	if len(id) <= dnsLabelMaxLength {
+		return id
+	}
+
+	// TODO: get original root
```

Comment:
Which API would be the best one to get original root node, 
if we fetch it with intention to re-hash using different hash function, while keeping everything else intact (codec, links, metadata etc)

Reply:
`api.ResolveNode`